jsonnet/components/admission-webhook.libsonnet

```diff
@@ -0,0 +1,208 @@
+    podDisruptionBudget: {
+      apiVersion: 'policy/v1',
+      kind: 'PodDisruptionBudget',
+      metadata: {
+        name: aw._config.name,
+        namespace: aw._config.namespace,
+        labels: aw._config.commonLabels,
+      },
+      spec: {
+        minAvailable: 1,
+        selector: {
+          matchLabels: aw._config.selectorLabels,
+        },
+
+      },
+    },
+
+    prometheusRuleValidatingWebhook: {
+      apiVersion: 'admissionregistration.k8s.io/v1',
+      kind: 'ValidatingWebhookConfiguration',
+      metadata: {
+        name: 'prometheusrules.openshift.io',
+        labels: {
+          'app.kubernetes.io/component': 'controller',
+          'app.kubernetes.io/name': 'prometheus-operator',
+        },
+        annotations: {
+          'service.beta.openshift.io/inject-cabundle': 'true',
+        },
+      },
+      webhooks: [
+        {
+          name: 'prometheusrules.openshift.io',
+          rules: [
+            {
+              apiGroups: ['monitoring.coreos.com'],
+              apiVersions: ['v1'],
+              operations: ['CREATE', 'UPDATE'],
+              resources: ['prometheusrules'],
+              scope: 'Namespaced',
+            },
+          ],
+          clientConfig: {
+            service: {
+              namespace: 'openshift-monitoring',
+              name: 'prometheus-operator-admission-webhook',
+              port: 8443,
+              path: '/admission-prometheusrules/validate',
+            },
+          },
+          admissionReviewVersions: ['v1'],
+          sideEffects: 'None',
+          timeoutSeconds: 5,
+          failurePolicy: 'Ignore',
+        },
+      ],
+    },
+
+    alertmanagerConfigValidatingWebhook: {
+      apiVersion: 'admissionregistration.k8s.io/v1',
+      kind: 'ValidatingWebhookConfiguration',
+      metadata: {
+        name: 'alertmanagerconfigs.openshift.io',
+        labels: {
+          'app.kubernetes.io/component': 'controller',
+          'app.kubernetes.io/name': 'prometheus-operator',
+        },
+        annotations: {
+          'service.beta.openshift.io/inject-cabundle': 'true',
+        },
+      },
+      webhooks: [
+        {
+          name: 'alertmanagerconfigs.openshift.io',
+          rules: [
+            {
+              apiGroups: ['monitoring.coreos.com'],
+              apiVersions: ['v1alpha1'],
+              operations: ['CREATE', 'UPDATE'],
+              resources: ['alertmanagerconfigs'],
+              scope: 'Namespaced',
+            },
+          ],
+          clientConfig: {
+            service: {
+              namespace: 'openshift-monitoring',
+              name: 'prometheus-operator-admission-webhook',
+              port: 8443,
+              path: '/admission-alertmanagerconfigs/validate',
+            },
+          },
+          admissionReviewVersions: ['v1'],
+          sideEffects: 'None',
+          timeoutSeconds: 5,
+          failurePolicy: 'Ignore',
```

Comment:
I wonder can these policies be changed now to `fail`? 🤔

Reply:
yes this would be the idea but I want to do it in a follow-up PR :)